### ISSUE TITLE
auto-install can't pick the right artifact for the platform

### ISSUE THREAD

**Issue by amogorkon** [amogorkon/justuse#402]
on jupyter, ubuntu:

```
spam = use("spam", version="0.6.1", hashes={'X䂦嶲驹瑐㮝嬫邌舱衹糩濷躡锠浭陜ȇ䠐', 'O绉䫳鋾䮭䃰襩囫Ğ骪ȟ怚爯䡅鮃軾趶眽', 'X绡蟙婵乙䗅庯婽跷㨢磟Ů粩碗孑䛘炤聄'}, modes=use.auto_install)

Last executed at 2021-11-23 21:27:21 in 20.88s

---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-15-ff7abfa03cd7> in <module>
----> 1 spam = use("spam", version="0.6.1", hashes={'X䂦嶲驹瑐㮝嬫邌舱衹糩濷躡锠浭陜ȇ䠐', 'O绉䫳鋾䮭䃰襩囫Ğ骪ȟ怚爯䡅鮃軾趶眽', 'X绡蟙婵乙䗅庯婽跷㨢磟Ů粩碗孑䛘炤聄'}, modes=use.auto_install)

~/anaconda3/lib/python3.8/site-packages/use/main.py in __call__(self, *args, **kwargs)
    101     def __call__(self, *args, **kwargs):
    102         with self.__condition:
--> 103             return self.__implementation(*args, **kwargs)
    104 
    105 

~/anaconda3/lib/python3.8/site-packages/use/tools.py in wrapper(*args, **kw)
     40         if len(args) == 1:
     41             args = args + (None,)
---> 42         return dispatcher.dispatch(args[1].__class__)(*args, **kw)
     43 
     44     wrapper.register = dispatcher.register

~/anaconda3/lib/python3.8/site-packages/use/main.py in _use_str(self, name, version, hash_algo, hashes, default, modes)
    706         """
    707         package_name, module_name = _parse_name(name)
--> 708         return self._use_package(
    709             name=name,
    710             package_name=package_name,

~/anaconda3/lib/python3.8/site-packages/use/main.py in _use_package(self, name, package_name, module_name, version, hashes, modes, default, hash_algo, user_msg)
    771             "sys_version": Version(".".join(map(str, sys.version_info[0:3]))),
    772         }
--> 773         result = buffet_table(case, kwargs)
    774         assert result
    775 

~/anaconda3/lib/python3.8/site-packages/use/buffet_old.py in buffet_table(case, kwargs)
     38     log.info("case_func = '%s' %s", case_func.__qualname__, case_func)
     39     log.info("kwargs = %s", repr(kwargs))
---> 40     result = case_func()
     41     log.info("result = %s", repr(result))
     42     return result

~/anaconda3/lib/python3.8/site-packages/use/buffet_old.py in <lambda>()
     30         (0, 1, 1, 1): lambda: pimp._pebkac_no_version(**kwargs),
     31         (1, 0, 1, 1): lambda: pimp._pebkac_no_hash(**kwargs),
---> 32         (1, 1, 0, 1): lambda: pimp._auto_install(**kwargs),
     33         (1, 1, 1, 0): lambda: pimp._ensure_version(pimp._import_public_no_install(**kwargs), **kwargs),
     34         (1, 1, 1, 1): lambda: pimp._auto_install(

~/anaconda3/lib/python3.8/site-packages/use/pimp.py in _auto_install(name, func, version, hash_algo, package_name, module_name, **kwargs)
    378 
    379     if not query or not _ensure_path(query["artifact_path"]).exists():
--> 380         query = _find_or_install(package_name, version)
    381 
    382     artifact_path = _ensure_path(query["artifact_path"])

~/anaconda3/lib/python3.8/site-packages/use/pimp.py in _find_or_install(name, version, artifact_path, url, out_info, force_install)
    626 
    627         # If we get here, the venv/pip setup is required.
--> 628         output = _process(
    629             python_exe,
    630             "-m",

~/anaconda3/lib/python3.8/site-packages/use/pimp.py in _process(env, *argv)
    459     if o.returncode == 0:
    460         return o
--> 461     raise RuntimeError(  # cov: exclude
    462         "\x0a".join(
    463             (

RuntimeError: 
Problem running--command exited with non-zero: 1
/home/thorsten/anaconda3/bin/python -m pip --disable-pip-version-check --no-color install --pre --root / --prefix /home/thorsten/.justuse-python/venv/spam/0.6.1 --progress-bar ascii --prefer-binary --exists-action i --ignore-installed --no-warn-script-location --force-reinstall --no-warn-conflicts /home/thorsten/.justuse-python/packages/spam-0.6.1-cp36-cp36m-manylinux2014_x86_64.whl
---[  Errors  ]---
ERROR: spam-0.6.1-cp36-cp36m-manylinux2014_x86_64.whl is not a supported wheel on this platform.


Arguments to subprocess.run(**setup):
{ 'args': [ '/home/thorsten/anaconda3/bin/python',
            '-m',
            'pip',
            '--disable-pip-version-check',
            '--no-color',
            'install',
            '--pre',
            '--root',
            '/',
            '--prefix',
            '/home/thorsten/.justuse-python/venv/spam/0.6.1',
            '--progress-bar',
            'ascii',
            '--prefer-binary',
            '--exists-action',
            'i',
            '--ignore-installed',
            '--no-warn-script-location',
            '--force-reinstall',
            '--no-warn-conflicts',
            '/home/thorsten/.justuse-python/packages/spam-0.6.1-cp36-cp36m-manylinux2014_x86_64.whl'],
  'bufsize': 1024,
  'capture_output': True,
  'check': False,
  'close_fds': True,
  'encoding': 'ISO-8859-1',
  'env': { 'BROWSER': 'firefox',
           'CLICOLOR': '1',
           'COLORTERM': 'truecolor',
           'CONDA_DEFAULT_ENV': 'base',
           'CONDA_EXE': '/home/thorsten/anaconda3/bin/conda',
           'CONDA_PREFIX': '/home/thorsten/anaconda3',
           'CONDA_PROMPT_MODIFIER': '(base) ',
           'CONDA_PYTHON_EXE': '/home/thorsten/anaconda3/bin/python',
           'CONDA_SHLVL': '1',
           'DBUS_SESSION_BUS_ADDRESS': 'unix:path=/run/user/1000/bus',
           'DESKTOP_SESSION': 'Lubuntu',
           'DISPLAY': ':0',
           'GDK_BACKEND': 'x11',
           'GIT_PAGER': 'cat',
           'GPG_AGENT_INFO': '/run/user/1000/gnupg/S.gpg-agent:0:1',
           'GTK_CSD': '0',
           'GTK_OVERLAY_SCROLLING': '0',
           'HOME': '/home/thorsten',
           'JPY_PARENT_PID': '2039',
           'LANG': 'de_DE.UTF-8',
           'LC_ADDRESS': 'de_DE.UTF-8',
           'LC_IDENTIFICATION': 'de_DE.UTF-8',
           'LC_MEASUREMENT': 'de_DE.UTF-8',
           'LC_MONETARY': 'de_DE.UTF-8',
           'LC_NAME': 'de_DE.UTF-8',
           'LC_NUMERIC': 'de_DE.UTF-8',
           'LC_PAPER': 'de_DE.UTF-8',
           'LC_TELEPHONE': 'de_DE.UTF-8',
           'LC_TIME': 'de_DE.UTF-8',
           'LESSCLOSE': '/usr/bin/lesspipe %s %s',
           'LESSOPEN': '| /usr/bin/lesspipe %s',
           'LOGNAME': 'thorsten',
           'LS_COLORS': 'rs=0:di=01;34:ln=01;36:mh=00:pi=40;33:so=01;35:do=01;35:bd=40;33;01:cd=40;33;01:or=40;31;01:mi=00:su=37;41:sg=30;43:ca=30;41:tw=30;42:ow=34;42:st=37;44:ex=01;32:*.tar=01;31:*.tgz=01;31:*.arc=01;31:*.arj=01;31:*.taz=01;31:*.lha=01;31:*.lz4=01;31:*.lzh=01;31:*.lzma=01;31:*.tlz=01;31:*.txz=01;31:*.tzo=01;31:*.t7z=01;31:*.zip=01;31:*.z=01;31:*.dz=01;31:*.gz=01;31:*.lrz=01;31:*.lz=01;31:*.lzo=01;31:*.xz=01;31:*.zst=01;31:*.tzst=01;31:*.bz2=01;31:*.bz=01;31:*.tbz=01;31:*.tbz2=01;31:*.tz=01;31:*.deb=01;31:*.rpm=01;31:*.jar=01;31:*.war=01;31:*.ear=01;31:*.sar=01;31:*.rar=01;31:*.alz=01;31:*.ace=01;31:*.zoo=01;31:*.cpio=01;31:*.7z=01;31:*.rz=01;31:*.cab=01;31:*.wim=01;31:*.swm=01;31:*.dwm=01;31:*.esd=01;31:*.jpg=01;35:*.jpeg=01;35:*.mjpg=01;35:*.mjpeg=01;35:*.gif=01;35:*.bmp=01;35:*.pbm=01;35:*.pgm=01;35:*.ppm=01;35:*.tga=01;35:*.xbm=01;35:*.xpm=01;35:*.tif=01;35:*.tiff=01;35:*.png=01;35:*.svg=01;35:*.svgz=01;35:*.mng=01;35:*.pcx=01;35:*.mov=01;35:*.mpg=01;35:*.mpeg=01;35:*.m2v=01;35:*.mkv=01;35:*.webm=01;35:*.ogm=01;35:*.mp4=01;35:*.m4v=01;35:*.mp4v=01;35:*.vob=01;35:*.qt=01;35:*.nuv=01;35:*.wmv=01;35:*.asf=01;35:*.rm=01;35:*.rmvb=01;35:*.flc=01;35:*.avi=01;35:*.fli=01;35:*.flv=01;35:*.gl=01;35:*.dl=01;35:*.xcf=01;35:*.xwd=01;35:*.yuv=01;35:*.cgm=01;35:*.emf=01;35:*.ogv=01;35:*.ogx=01;35:*.aac=00;36:*.au=00;36:*.flac=00;36:*.m4a=00;36:*.mid=00;36:*.midi=00;36:*.mka=00;36:*.mp3=00;36:*.mpc=00;36:*.ogg=00;36:*.ra=00;36:*.wav=00;36:*.oga=00;36:*.opus=00;36:*.spx=00;36:*.xspf=00;36:',
           'LXQT_DEFAULT_OPENBOX_CONFIG': '/etc/xdg/xdg-Lubuntu/openbox/lxqt-rc.xml',
           'LXQT_SESSION_CONFIG': 'session',
           'MPLBACKEND': 'module://ipykernel.pylab.backend_inline',
           'OLDPWD': '/media',
           'OPENBLAS_NUM_THREADS': '1',
           'PAGER': 'cat',
           'PATH': '/home/thorsten/anaconda3/bin:/home/thorsten/anaconda3/condabin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin',
           'PWD': '/media/sf_Dropbox',
           'QT_ACCESSIBILITY': '1',
           'QT_PLATFORM_PLUGIN': 'lxqt',
           'QT_QPA_PLATFORMTHEME': 'lxqt',
           'SAL_USE_VCLPLUGIN': 'qt5',
           'SHELL': '/bin/bash',
           'SHLVL': '1',
           'SSH_AGENT_PID': '1014',
           'SSH_AUTH_SOCK': '/tmp/ssh-Gnof51AZQfJA/agent.931',
           'TERM': 'xterm-color',
           'USER': 'thorsten',
           'VTE_VERSION': '6003',
           'XAUTHORITY': '/home/thorsten/.Xauthority',
           'XDG_CACHE_HOME': '/home/thorsten/.cache',
           'XDG_CONFIG_DIRS': '/etc/xdg/xdg-Lubuntu:/etc/xdg:/etc:/usr/share',
           'XDG_CONFIG_HOME': '/home/thorsten/.config',
           'XDG_CURRENT_DESKTOP': 'LXQt',
           'XDG_DATA_DIRS': '/usr/share/Lubuntu:/usr/local/share:/usr/share:/var/lib/snapd/desktop',
           'XDG_DATA_HOME': '/home/thorsten/.local/share',
           'XDG_MENU_PREFIX': 'lxqt-',
           'XDG_RUNTIME_DIR': '/run/user/1000',
           'XDG_SEAT': 'seat0',
           'XDG_SEAT_PATH': '/org/freedesktop/DisplayManager/Seat0',
           'XDG_SESSION_CLASS': 'user',
           'XDG_SESSION_DESKTOP': '',
           'XDG_SESSION_ID': '1',
           'XDG_SESSION_PATH': '/org/freedesktop/DisplayManager/Session0',
           'XDG_SESSION_TYPE': 'x11',
           'XDG_VTNR': '1',
           '_': '/home/thorsten/anaconda3/bin/jupyter',
           '_CE_CONDA': '',
           '_CE_M': ''},
  'errors': 'ISO-8859-1',
  'executable': PosixPath('/home/thorsten/anaconda3/bin/python'),
  'input': '',
  'shell': False,
  'text': True,
  'timeout': 45000}
---[  STDOUT  ]---

---[  STDERR  ]---
ERROR: spam-0.6.1-cp36-cp36m-manylinux2014_x86_64.whl is not a supported wheel on this platform.

```

**Comment by greyblue9** [comment]
> 

Hoping this will be fixed now; if you run 3.6 to 3.8 on Ubuntu x86_64, it should at least install the package; otherwise, it should raise a more informative error rather than trying the first version in the list

**Comment by amogorkon** [comment]
spam sucks. the issue seems fixed in general. good enough.
